indenter.cpp

```diff
@@ -0,0 +1,51 @@
+#include "indenter.hpp"
```

Comment:
This code, and the .hpp file are the big guts of the changes.   They are what enables me to do clean output to file
(or any iostream) which preserves the existing formatting provided by the custom file classes.

They work by acting as a buffering streambuf and insert themselves on the output streams between the  incoming data and underlying original buffer.  When data comes into the buffer, it performs the formatting required based
on the state as updated by the manipulators and then on a newline character or stream flush, output any buffered data to the underlying original buf after modification.